### PR TITLE
Update cal-heatmap.js with d3 support for AMD/UMD 

### PR DESCRIPTION
**Pull request by t2k**
Add d3 support for amd/umd

### CODE REVIEW

**Comment by kamisama** [comment]
There's a lot of unrelated changes in the commit, since you change all tab with a space. Can you save the file again with tab instead of spaces ?

**Comment by t2k** [comment]
I just wrapped your final output file so I could use it quickly in a project.  Let me take a look at a grunt umd plugin and I'll integrate it directly into your grunt build task.  I'm more familiar with gulp, but have used grunt before, s  I'll take a look see...

**Comment by t2k** [comment]
Hi,  I'd like to try to add the grund-umd task to your grunt build process.   Taking a fresh start, I cloned your repo and cannot seem to successfully run  npm install in my Ubuntu 14.04 environment.  There are numerous warnings about outdated versions of node and npm and finally errors installing the 'karma' package.    Can you refresh the package.json perhaps?

**Comment by kamisama** [comment]
I'll take a look

**Comment by t2k** [comment]
Latest commit adds the grunt-umd task to grunt build.  See if that works... I created a build/ folder but still output .js files to ./ to avoid breaking changes.

**Comment by t2k** [comment]
not sure if you had a chance to look this over.  Should be good to go, I just integrated grunt-umd into the grunt build process.  The update adds proper dependency support for d3 in modular build stacks using AMD or commonJS  plus support for window globals.  I ref my fork for my AMD projects that use cal-heatmap in backbone.marionette views.

**Comment by kamisama** [comment]
Sorry, I haven't touched any js code for months, so it'll take me some time to be up-to-date on this project again.

If I understand the PR correctly, you are removing and delegating the AMD support hardcoded in the code to the grunt-umd task ?

And again, if I read the changes correctly, the new code is now supporting CommonJS ?

What are the breaking changes you're talking about when justifying using the build folder ?
Why was the css moved ?

**Comment by t2k** [comment]
Yes, the grunt-umd task is configured correctly to add AMD, CommonJS and window global support with the d3 dependencies.

I was considering outputting the .js files to a /dist folder but decided against doing so as that would have caused a breaking change if somebody updated via Bower etc.  It's not a breaking change in a sense that any code was changing or worked differently, just the location would change.

Regards,  --Ted

**Comment by kamisama** [comment]
Why don't you output the final file to the root folder ? In any case, the files in the root are built by grunt.

Can you move back the css file to the root ?

Can you do a rebase with the latest master ? No big changes, but last version is 3.5.1.

**Comment by infacq** [comment]
my problem is solved with @t2k 's library

**Comment by t2k** [comment]
I do output to the root folder.  I only suggested that the final output from the build process might be better outputted to a /dist folder.    ie  src  ==> buiild ==> dist

I didn't because it would cause a minor breaking change.  So I did leave the UMD task to output to the root.  You should be safe to apply this.

File: Gruntfile.js [hunk 1]
```diff
@@ -86,13 +89,34 @@ module.exports = function(grunt) {
 				}
 			},
 			lint: {
-				files: "src/*.js",
+				files: "build/*.js",
```
Comment (kamisama) [inline Gruntfile.js]: Leave that to `src`, to enable linting while developing.

File: Gruntfile.js [hunk 2]
```diff
@@ -86,13 +89,34 @@ module.exports = function(grunt) {
...
+                    indent: 2, // optional (defaults to 2), indent source code. Accepts strings as well
+                    deps: { // optional
+                        'default': ['d3'],
+                        amd: ['d3'],
+                        cjs: ['d3'],
+                        global: ['d3']
+                    }
+                }
+            }
+        }
```
Comment (kamisama) [inline Gruntfile.js]: Use double quote, like the rest of the file

File: Gruntfile.js [hunk 3]
```diff
@@ -102,15 +126,19 @@ module.exports = function(grunt) {
...
+    grunt.loadNpmTasks("grunt-contrib-clean");
 
     // TO RUN BEFORE COMMIT
     // ====================
-    grunt.registerTask("quick-build", ["csslint", "jshint"]);
+    grunt.registerTask("quick-build", ["clean:build","csslint", "jshint"]);
 
     // Full build without version bump
-    grunt.registerTask("build", ["concat", "qunit", "csslint", "jshint", "uglify"]);
+    grunt.registerTask("build", ["clean:build","umd:all","concat", "uglify"]);
```
Comment (kamisama) [inline Gruntfile.js]: Leave a space after `,`

**Comment by wa0x6e** [comment]
Added in v4

**Pull request closed without merging**